presto-geospatial/src/main/java/com/facebook/presto/plugin/geospatial/BingTileFunctions.java

```diff
@@ -190,6 +193,136 @@ public static Block bingTilesAround(
         return blockBuilder.build();
     }
 
+    @Description("Given a (longitude, latitude) point, returns the surrounding Bing tiles " +
+            "at the specified zoom level covering minimum radiusInKm")
+    @ScalarFunction("bing_tiles_around")
+    @SqlType("array(" + BingTileType.NAME + ")")
+    public static Block bingTilesAround(
+            @SqlType(StandardTypes.DOUBLE) double latitude,
+            @SqlType(StandardTypes.DOUBLE) double longitude,
+            @SqlType(StandardTypes.INTEGER) long zoomLevel,
+            @SqlType(StandardTypes.DOUBLE) double radiusInKm)
+    {
+        checkLatitude(latitude, LATITUDE_OUT_OF_RANGE);
+        checkLongitude(longitude, LONGITUDE_OUT_OF_RANGE);
+        checkZoomLevel(zoomLevel);
+
+        long mapSize = mapSize(toIntExact(zoomLevel));
+        long maxTileIndex = (mapSize / TILE_PIXELS) - 1;
+
+        long tileY = longitudeToTileY(latitude, mapSize);
+        long tileX = longitudeToTileX(longitude, mapSize);
+
+        // Find top, bottom, left and right tiles from center of circle
+        double[] topLatLong = GeoFunctions.addDistanceToLatitudeLongitude(latitude, longitude, radiusInKm, 0);
```

Comment:
I feel that using generic function here causes good amount of unnecessary computation and adds to CPU cost. Is there a way to avoid redundant computation? E.g. we know that longitude for top and bottom points is the same as longitude of the center point. Similarly, latitude for left and right points is the same is latitude of the center point.

Reply:
I was using a different formula before but that didnt seem accurate enough (https://stackoverflow.com/questions/7477003/calculating-new-longitude-latitude-from-old-n-meters). Let me know if that works.

Reply:
@avinashidnani I'm not suggesting to change the formula, but to look into how to avoid repeating the same computation.